code/modules/mob/living/carbon/human/powers/check_skills.dm

```diff
@@ -0,0 +1,73 @@
+/mob/living/carbon/human/verb/checkSkills()
+	set name = "Check Skills"
+	set category = "IC"
+	set src = usr
+
+	src.skills.tgui_interact(src)
+
+/datum/skills/ui_state(mob/user)
+	return GLOB.always_state
+
+/datum/skills/tgui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "SkillsMenu", "[owner]'s skills")
+		ui.open()
+		ui.set_autoupdate(FALSE)
+
+/datum/skills/ui_static_data(mob/user)
+	var/list/data = list()
+
+	data["admin"] = check_rights(R_VAREDIT, FALSE)
+
+	return data
+
+/datum/skills/ui_data(mob/user)
+	var/list/data = list()
+
+	data["skillset_name"] = name
+	data["owner"] = owner
+
+
+	if(!skills)
+		data["skills"] = null
+		return data // they have no skills =  show the null skills menu
+
+	var/list/skills_data_list = list()
+
+	for(var/skillname as anything in skills)
+		var/datum/skill/skilldatum = get_skill(skillname)
+
+		skills_data_list += list(list(
+			"name" = skilldatum.readable_skill_name,
```

Comment:
```suggestion
			"name" = skilldatum.readable_skill_name ? skilldatum.readable_skill_name : skilldatum.skill_name
```
then we don't need to double up when readable_skill_name is just the same as skill_name, as in 90% of cases there